src/controllers/likes.ts

```diff
@@ -2,83 +2,88 @@ import { db } from '../database/db';
 import { asyncHandler } from '../middleware';
 import { cleanseData, ErrorResponse } from '../utils';
 
-/**
- * @desc    Get all likes (including profile information) for a listing
- * @route   GET /api/listings/:listing_id/likes
- * @desc    Get all likes (including listing information) for a user
- * @route   GET /api/users/:user_id/likes
- * @access  Public
- */
-export const getLikes = asyncHandler(async (req, res, next) => {
-    if (req.params.listing_id) {
-        // return 404 error response if listing not found or soft deleted
-        await db.one('SELECT * FROM listingview WHERE listing_id = $1', req.params.listing_id);
-
-        const likes = await db.manyOrNone('SELECT * FROM listinglike NATURAL JOIN profile WHERE listing_id = $1', req.params.listing_id);
-
-        return res.status(200).json({
-            success: true,
-            data: likes,
-        });
-    }
-
-    if (req.params.user_id) {
-        // return 404 error response if user not found
-        await db.one('SELECT * FROM loginuser WHERE user_id = $1', req.params.user_id);
-
-        const likes = await db.manyOrNone(
-            'SELECT * FROM listinglike ll JOIN listing ls ON ll.listing_id = ls.listing_id WHERE user_id = $1',
-            req.params.user_id,
-        );
-
-        return res.status(200).json({
-            success: true,
-            data: likes,
-        });
-    }
-
-    return next(new ErrorResponse('Invalid route', 404));
-});
+export class LikeController {
+    constructor() {}
+    /**
+     * @desc    Get all likes (including profile information) for a listing
+     * @route   GET /api/listings/:listing_id/likes
+     * @desc    Get all likes (including listing information) for a user
+     * @route   GET /api/users/:user_id/likes
+     * @access  Public
+     */
+    getLikes = asyncHandler(async (req, res, next) => {
```

Comment:
Can remove `asyncHandler` here as it already wraps this method in the route file.
```suggestion
    getLikes = async (req, res, next) => {
```